src/filters/oidc/jwks_storage.cc

```diff
@@ -0,0 +1,38 @@
+#include "src/filters/oidc/jwks_storage.h"
+
+namespace authservice {
+namespace filters {
+namespace oidc {
+
+NonPermanentJwksStorageImpl::JwksFetcher::JwksFetcher(
+    JwksStorage* parent, common::http::ptr_t http_ptr,
+    const std::string& jwks_uri, std::chrono::seconds duration,
+    boost::asio::io_context& io_context)
+    : parent_(parent),
+      jwks_uri_(jwks_uri),
+      http_ptr_(http_ptr),
+      ioc_(io_context),
+      duration_(duration),
+      timer_(ioc_, duration_) {
+  timer_.async_wait(
+      [this](const boost::system::error_code& ec) { this->request(ec); });
+}
+
+void NonPermanentJwksStorageImpl::JwksFetcher::request(
+    const boost::system::error_code&) {
+  boost::asio::spawn(ioc_, [this](boost::asio::yield_context yield) {
+    auto resp = http_ptr_->Get(jwks_uri_, {}, "", "", "", ioc_, yield);
+    const auto& fetched_jwks = resp->body();
+    // TODO(shikugawa): Prevent lock with same jwks.
+    // TODO(shikugawa): Support PEM.
```

Comment:
not sure if understand. what's the PEM used in this case? I am not aware of any keys that put PEM values in JWK or you mean something else? any link for the reference/example?

Reply:
This is not needed here. delete them.